tube/etl/indexers/aggregation/parser.py

```diff
@@ -106,7 +105,41 @@ def get_parent_props(self):
                 )
         return list_nodes
 
+    def add_program_name_to_parent(self):
+        """
+        In case program name is not in self.mapping["parent_props"] while the root node is project. We must add that field
+        :return:
+        """
+        found_program = -1
+        i = -1
+        for path in self.mapping["parent_props"]:
+            p = path.get("path")
+            i += 1
+            if p.startswith("program"):
+                found_program = i
+        if found_program == -1:
+            self.mapping["parent_props"].append(
+                {"path": "programs[{PROGRAM_N}:name]".format(PROGRAM_N=PROGRAM_NAME)}
+            )
+            found_program = len(self.mapping["parent_props"]) - 1
+        program_path = self.mapping["parent_props"][found_program]
+        program_path_val = program_path["path"]
+        if program_path_val.find(PROGRAM_NAME) == -1:
+            separator = "" if program_path_val.find("[]") > 0 else ","
+            program_path["path"] = "{}{SEP}{PROGRAM_N}:name]".format(
+                program_path_val[: len(program_path) - 1],
+                SEP=separator,
+                PROGRAM_N=PROGRAM_NAME,
+            )
+
     def get_host_props(self):
+        if self.root == "project":
```

Comment:
i'd argue that a mapping with `root = program` should work as well. the authz path should be `/programs/<program name>`. maybe that's a different ticket though

Reply:
yeah, it should be another ticket :)

Reply:
i created PXP-6511